os/binfmt/libelf/libelf_addrenv.c

```diff
@@ -200,9 +214,20 @@ int elf_addrenv_alloc(FAR struct elf_loadinfo_s *loadinfo, size_t textsize, size
 	 */
 	loadinfo->rosize += loadinfo->datasize - loadinfo->binp->bsssize;
 
+#ifdef CONFIG_ARMV7M_MPU
+	/* ARMV7M requires MPU region size to be a power of two */
 	loadinfo->textsize = 1 << mpu_log2regionceil(0, loadinfo->textsize);
 	loadinfo->rosize = 1 << mpu_log2regionceil(0, loadinfo->rosize);
 	datamemsize = 1 << mpu_log2regionceil(0, datamemsize);
+#elif CONFIG_ARMV8M_MPU
+	/* ARMV8M requires MPU region size to be aligned to 32 bytes */
+	loadinfo->textsize = MPU_ALIGN_UP(loadinfo->textsize);
+	loadinfo->rosize = MPU_ALIGN_UP(loadinfo->rosize);
+	datamemsize = MPU_ALIGN_UP(datamemsize);
+#else
+#error "Unknown MPU version. Expected either ARMV7M or ARMV8M"
```

Comment:
We can use binary loading regardless of mpu, even it is not a good way. So I think this can be a problem.

Reply:
This is true. But there are many places where MPU is tightly coupled to APP separation. 
If we need to make them independent, then we better refractor the code properly. 
Please let me know if this is required. We will do it in a separate PR.